.github/workflows/gradle_jvm_app_release.yml

```diff
@@ -0,0 +1,128 @@
+name: Release Docker Version
+
+on:
+  workflow_call:
+    inputs:
+      GH_CI_USER:
+        description: 'User for GitHub auth'
+        required: true
+        type: string
+      JAVA_VERSION:
+        required: false
+        type: string
+        default: 17
+    secrets:
+      GH_CI_PAT:
+        description: 'Token password for GitHub auth'
+        required: true
+
+      # Registry arguments.
+      # Must supply set of arguments for at least 1 registry.
+
+      # Artifact Registry arguments
+      ARTIFACT_REGISTRY:
+        description: 'Artifact Registry address to which to publish (leave blank to not publish)'
+        required: false
+      ARTIFACT_REGISTRY_JSON_KEY:
+        description: 'Key for publishing to Artifact Registry'
+        required: false
+      # Container Registry arguements
+      CONTAINER_REGISTRY:
+        description: 'Container Registry address to which to publish (leave blank to not publish)'
+        required: false
+      CONTAINER_REGISTRY_JSON_KEY:
+        description: 'Key for publishing to Container Registry'
+        required: false
+
+env:
+  GOPRIVATE: ${{ inputs.GOPRIVATE }}
+
+jobs:
+  push:
+    #
+    # Build the Docker image artifact and deliver it.
+    #
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout code needed to build docker image.
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: ${{ inputs.JAVA_VERSION }}
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+      - name: Execute Gradle Assemble
+        run: ./gradlew assemble
```

Comment:
I’m not really familiar with the Java/Kotlin build world, but this reminds me of our old days when we were building go binaries in the ci then just adding them in the dockerfile when really the dockerfile should be in control of the build entirely while any CI steps are just testing and linting.

@morris-j Is this a pattern we want to fully standardize for all "gradle_jvm" CI? My intuition says that the dockerfile should be responsible for building the jar, while CI should be responsible for running tests/lints, but if `./gradlew assemble` is just a super standard thing that is expected to be run for any possible app like this and maybe is even a pre-requirement for tests to run, then I could see this being more ok. The dockerfile at that point would feel a little rigid to me since it would only work if the files `./gradlew assemble` produces are in the right spot, essentially tying the dockerfile to this CI process instead of being more portable. But maybe I'm misunderstanding or missing context for this type of build.

Reply:
@23caterpie This is matching up with what was done in the previous CI logic. If a more Docker specific setup is desired the project supports more Docker specific build options as defined here [https://ktor.io/docs/docker.html](https://ktor.io/docs/docker.html) its just not something we have used before.

This may make it more docker generic but may also have the side effect of making it more ktor specific. I can't say for sure what is better or more generic as I am not overly familiar with these workflows.

Reply:
Ok. I'm fine with the workflow as it is as long as a Java/Kotlin expert (you) is fine with it. If we need/want to update this workflow in the future it's very easy. It might change some pre-requirements of the build, but I don't think we are exactly going to have cross team users having those kind of issues with this workflow.

Reply:
I think sticking with the basic setup this first time around will be good. We can work through other iterations as needed